<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="Align-GVGD (A-GVGD) is a method to predict the impact of
    missense substitutions based on the properties of amino acid side
    chains and protein multiple sequence alignments
    &lt;doi:10.1136/jmg.2005.033878&gt;. A-GVGD is an extension of the original
    Grantham distance to multiple sequence alignments. This package
    provides an alternative R implementation to the web version found on
    &lt;http://agvgd.hci.utah.edu/&gt;.">
<title>An R Implementation of the Align-GVGD (A-GVGD) Method • agvgd</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png">
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="An R Implementation of the Align-GVGD (A-GVGD) Method">
<meta property="og:description" content="Align-GVGD (A-GVGD) is a method to predict the impact of
    missense substitutions based on the properties of amino acid side
    chains and protein multiple sequence alignments
    &lt;doi:10.1136/jmg.2005.033878&gt;. A-GVGD is an extension of the original
    Grantham distance to multiple sequence alignments. This package
    provides an alternative R implementation to the web version found on
    &lt;http://agvgd.hci.utah.edu/&gt;.">
<meta property="og:image" content="https://maialab.org/agvgd/logo.svg">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">agvgd</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header">
<img src="logo.svg" class="logo" alt=""><h1 id="agvgd-">agvgd <a class="anchor" aria-label="anchor" href="#agvgd-"></a>
</h1>
</div>
<!-- badges: start -->
<!-- badges: end -->
<p>The R package <a href="https://maialab.org/agvgd">agvgd</a> provides an R implementation of the Align-GVGD<sup><a href="#1">1</a>,<a href="#2">2</a></sup> (A-GVGD) method.</p>
<p>A-GVGD combines multiple sequence alignment of orthologous sequences with the Grantham distance<sup><a href="#3">3</a></sup> to classify missense variants, i.e. to distinguish human disease susceptibility missense changes from changes of little clinical significance.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<!--
Install `{agvgd}` from CRAN:

``` r
install.packages("agvgd")
```
-->
<p>You can install the development version of <a href="https://maialab.org/agvgd">agvgd</a> like so:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># install.packages("remotes")</span>
<span class="fu">remotes</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"maialab/agvgd"</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a>
</h2>
<p>Here is a minimal example using a dummy alignment constructed from an R matrix:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://maialab.org/agvgd">agvgd</a></span><span class="op">)</span>

<span class="co"># The alignment can be provided as a simple matrix:</span>
<span class="va">alignment</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span>
  <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'P'</span>, <span class="st">'M'</span>, <span class="st">'I'</span>,
    <span class="st">'P'</span>, <span class="st">'I'</span>, <span class="st">'I'</span>,
    <span class="st">'P'</span>, <span class="st">'L'</span>, <span class="st">'I'</span><span class="op">)</span>,
  nrow <span class="op">=</span> <span class="fl">3</span>,
  byrow <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span>

<span class="co"># In this example we will interrogate changes to the second position in the</span>
<span class="co"># alignment, i.e., changes to Methionine (M); therefore we set the position of</span>
<span class="co"># interest (poi) to 2.</span>
<span class="va">poi</span> <span class="op">&lt;-</span> <span class="fl">2</span>

<span class="co"># Here's a set of three possible missense changes (substitutions): Isoleucine</span>
<span class="co"># (Ile, I), Leucine (Leu, L), and Tryptophan (Trp, W):</span>
<span class="va">substitutions</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'I'</span>, <span class="st">'L'</span>, <span class="st">'W'</span><span class="op">)</span>

<span class="co"># agvgd package's main function is `agvgd()` :)</span>
<span class="fu"><a href="reference/agvgd.html">agvgd</a></span><span class="op">(</span>alignment <span class="op">=</span> <span class="va">alignment</span>, poi <span class="op">=</span> <span class="va">poi</span>, <span class="va">substitutions</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 3 × 5</span>
<span class="co">#&gt;   reference substitution    gv    gd prediction</span>
<span class="co">#&gt;   &lt;chr&gt;     &lt;chr&gt;        &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;     </span>
<span class="co">#&gt; 1 Met       Ile           14.3   0   C0        </span>
<span class="co">#&gt; 2 Met       Leu           14.3   0   C0        </span>
<span class="co">#&gt; 3 Met       Trp           14.3  60.4 C35</span></code></pre></div>
<p>This is another simple example but using one of the bundled alignments. This one is the alignment for the gene ATM. Let’s say you are interested in position 16 of the alignment, and that you would like to know the impact of all possible missense substitutions:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># `read_alignment()` either reads in one of the bundled alignments with</span>
<span class="co"># `{agvgd}` by gene name or directly from a FASTA file.</span>
<span class="va">alignment</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/read_alignment.html">read_alignment</a></span><span class="op">(</span>gene <span class="op">=</span> <span class="st">'ATM'</span><span class="op">)</span>

<span class="co"># For the sake of illustration, let us define a new alignment based on a narrow</span>
<span class="co"># region of the original alignment by focusing only on the first 30 positions.</span>
<span class="co"># Note: rows are protein sequences; columns are alignment positions.</span>
<span class="va">narrow_alignment</span> <span class="op">&lt;-</span> <span class="va">alignment</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">30</span><span class="op">]</span>

<span class="co"># poi: position of interest</span>
<span class="va">poi</span> <span class="op">&lt;-</span> <span class="fl">16</span>

<span class="co"># Print the alignment to the console and highlight the position of interest</span>
<span class="co"># (POI). NB: The POI is not highlighted in GitHub's README but it is on the R</span>
<span class="co"># console.</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">narrow_alignment</span>, poi <span class="op">=</span> <span class="va">poi</span><span class="op">)</span>
<span class="co">#&gt; Hsap_ATM_AAB65827.1  1 M-SLVLNDLLICCRQLEHDRATERKKEVEK</span>
<span class="co">#&gt; Mmus_ATM_NP_031525.2 1 M-SLALNDLLICCRQLEHDRATERRKEVDK</span>
<span class="co">#&gt; Sscr_ATM_AAT01608.1  1 M-SLALNDLLICCRQLEHDRATERRKAVEN</span>
<span class="co">#&gt; Mdom_ATM_IARC        1 M-SLALNDLLLCCRQLENDRATERRKEVEK</span>
<span class="co">#&gt; Ggal_ATM_edited      1 M-SLVLHDLLTCCRRLENERATERRNEIEN</span>
<span class="co">#&gt; Xlae_ATM_AAT72929.1  1 M-SLALHELLLCCRQIETDKATERKKEIVK</span>
<span class="co">#&gt; Drer_ATM_IARC_v2     1 M-SLALHELLVCCRGLENEKATERKKEVDR</span>
<span class="co">#&gt; Bflo_ATM_IARC        1 MTDLLTHDLRDCCCHLESDKVTERKKNAEK</span>
<span class="co">#&gt; Spur_ATM_ABY60856.1  1 MAEVLIP-LRTACGYLGSDKITERKKQIDI</span>

<span class="co"># You may use `amino_acids()` to get a vector of all the 20 standard amino acids</span>
<span class="co"># and hence easily get a vector of all possible substitutions:</span>
<span class="va">all_substitutions</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/amino_acids.html">amino_acids</a></span><span class="op">(</span><span class="op">)</span>

<span class="fu"><a href="reference/agvgd.html">agvgd</a></span><span class="op">(</span>alignment <span class="op">=</span> <span class="va">narrow_alignment</span>, poi <span class="op">=</span> <span class="va">poi</span>, <span class="va">all_substitutions</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 20 × 5</span>
<span class="co">#&gt;    reference substitution    gv    gd prediction</span>
<span class="co">#&gt;    &lt;chr&gt;     &lt;chr&gt;        &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;     </span>
<span class="co">#&gt;  1 Leu       Ser           4.86 142.  C65       </span>
<span class="co">#&gt;  2 Leu       Arg           4.86  97.6 C65       </span>
<span class="co">#&gt;  3 Leu       Leu           4.86   0   C0        </span>
<span class="co">#&gt;  4 Leu       Pro           4.86  95.4 C65       </span>
<span class="co">#&gt;  5 Leu       Thr           4.86  89.3 C65       </span>
<span class="co">#&gt;  6 Leu       Ala           4.86  93.7 C65       </span>
<span class="co">#&gt;  7 Leu       Val           4.86  29.6 C15       </span>
<span class="co">#&gt;  8 Leu       Gly           4.86 135.  C65       </span>
<span class="co">#&gt;  9 Leu       Ile           4.86   0   C0        </span>
<span class="co">#&gt; 10 Leu       Phe           4.86  21.3 C0        </span>
<span class="co">#&gt; 11 Leu       Tyr           4.86  33.0 C25       </span>
<span class="co">#&gt; 12 Leu       Cys           4.86 197.  C65       </span>
<span class="co">#&gt; 13 Leu       His           4.86  94.3 C65       </span>
<span class="co">#&gt; 14 Leu       Gln           4.86 109.  C65       </span>
<span class="co">#&gt; 15 Leu       Asn           4.86 149.  C65       </span>
<span class="co">#&gt; 16 Leu       Lys           4.86 102.  C65       </span>
<span class="co">#&gt; 17 Leu       Asp           4.86 168.  C65       </span>
<span class="co">#&gt; 18 Leu       Glu           4.86 134.  C65       </span>
<span class="co">#&gt; 19 Leu       Met           4.86  10.1 C0        </span>
<span class="co">#&gt; 20 Leu       Trp           4.86  60.5 C55</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="logo">Logo<a class="anchor" aria-label="anchor" href="#logo"></a>
</h2>
<p>The <a href="https://maialab.org/agvgd">agvgd</a> logo, <code>agvgd.png</code>, is a derivative work of an illustration of <a href="https://pdb101.rcsb.org/motm/206" class="external-link">“Globin Evolution”</a> by <a href="https://ccsb.scripps.edu/goodsell/" class="external-link">David S. Goodsell</a> and the <a href="https://www.rcsb.org/" class="external-link">RCSB PDB</a>, used under <a href="https://creativecommons.org/licenses/by/4.0/" class="external-link">CC-BY-4.0</a>. <code>agvgd.png</code> is licensed under CC-BY-4.0 by Ramiro Magno.</p>
</div>
<div class="section level2">
<h2 id="related-software">Related software<a class="anchor" aria-label="anchor" href="#related-software"></a>
</h2>
<ul>
<li>Align-GVGD (A-GVGD) by <a href="https://uofuhealth.utah.edu/huntsman/labs/tavtigian/" class="external-link">Sean Tavtigian</a>, provided as a web service and hosted by the <a href="https://healthcare.utah.edu/huntsmancancerinstitute/" class="external-link">Huntsman Cancer Institute</a> at <a href="http://agvgd.hci.utah.edu/" class="external-link uri">http://agvgd.hci.utah.edu/</a>.</li>
<li>Multivariate Analysis of Protein Polymorphism (MAPP)<sup><a href="#4">4</a></sup> by <a href="https://bdsi.anu.edu.au/people/professor-eric-stone" class="external-link">Eric Stone</a> that had the same ideas as A-GVGD but in a more sophisticated framework. This used to be provided as a Java standalone application: <a href="https://biology.anu.edu.au/research/groups/stone-group-quantitative-and-computational-biology/software" class="external-link uri">https://biology.anu.edu.au/research/groups/stone-group-quantitative-and-computational-biology/software</a>; however, the download link to MAPP has been down for a long while.</li>
<li>The <a href="https://cran.r-project.org/package=grantham" class="external-link">grantham</a> package.</li>
</ul>
</div>
<div class="section level2">
<h2 id="acknowledgements">Acknowledgements<a class="anchor" aria-label="anchor" href="#acknowledgements"></a>
</h2>
<p>The work on this package benefited greatly from the feedback of Professor Sean Tavtigian, and Dr. Russell Bell.</p>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<p><a id="1">1.</a> Tavtigian, S.V., Deffenbaugh, A. M., Yin, L., Judkins, T., Scholl, T., Samollow, P.B., de Silva, D., Zharkikh, A., Thomas, A. <em>Comprehensive statistical study of 452 BRCA1 missense substitutions with classification of eight recurrent substitutions as neutral</em>. Journal of Medical Genetics 43, 295–305 (2006). doi: <a href="https://doi.org/10.1136/jmg.2005.033878" class="external-link">10.1136/jmg.2005.033878</a>.</p>
<p><a id="2">2.</a> Mathe, E., Olivier, M., Kato, S., Ishioka, C., Hainaut, P., Tavtigian, S.V. <em>Computational approaches for predicting the biological effect of p53 missense mutations: a comparison of three sequence analysis based methods</em>. Nucleic Acids Research 34, 1317–1325 (2006). doi: <a href="https://doi.org/10.1093/nar/gkj518" class="external-link">10.1093/nar/gkj518</a>.</p>
<p><a id="3">3.</a> Grantham, R. <em>Amino acid difference formula to help explain protein evolution</em>. Science 185, 862–864 (1974). doi: <a href="https://doi.org/10.1126/science.185.4154.862" class="external-link">10.1126/science.185.4154.862</a>.</p>
<p><a id="4">4.</a> Stone, E. A., Sidow, Arend. <em>Physicochemical constraint violation by missense substitutions mediates impairment of protein function and disease severity</em>. Genome Research 15, 978–986 (2005). doi: <a href="https://doi.org/10.1101/gr.3804205" class="external-link">10.1101/gr.3804205</a>.</p>
</div>
</div>
  </main><aside class="col-md-3"><div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing agvgd</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>
<a href="https://rmagno.eu/" class="external-link">Ramiro Magno</a> <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0001-5226-3441" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>
<a href="http://iduarte.eu/" class="external-link">Isabel Duarte</a> <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0003-0060-2936" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Ana-Teresa Maia <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-0454-9207" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>
<a href="https://cintesis.eu/" class="external-link"><img src="reference/figures/cintesis.svg" height="24"></a> <br><small class="roles"> Copyright holder, funder </small>  </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by <a href="https://rmagno.eu/" class="external-link">Ramiro Magno</a>, <a href="http://iduarte.eu/" class="external-link">Isabel Duarte</a>, Ana-Teresa Maia, <a href="https://cintesis.eu/" class="external-link"><img src="reference/figures/cintesis.svg" height="24"></a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
