<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="This function implements the Align-GVGD (A-GVGD) method described in
Tavtigian et al. (2006).
A-GVGD combines multiple sequence alignment of orthologous sequences with the
Grantham distance to classify missense variants, i.e. to distinguish human
disease susceptibility missense changes from changes of little clinical
significance.
The biochemical variation at each alignment position is converted to a
Grantham Variation score (GV) and the difference between these properties and
those of the variant amino acid being assessed are calculated and a Grantham
Difference score generated (GD).  The predicted effect is classed as C0, C15,
C25, C35, C45, C55, or C65, with C65 most likely to interfere with function
and C0 least likely."><title>Align-GVGD (A-GVGD) — agvgd • agvgd</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Align-GVGD (A-GVGD) — agvgd"><meta property="og:description" content="This function implements the Align-GVGD (A-GVGD) method described in
Tavtigian et al. (2006).
A-GVGD combines multiple sequence alignment of orthologous sequences with the
Grantham distance to classify missense variants, i.e. to distinguish human
disease susceptibility missense changes from changes of little clinical
significance.
The biochemical variation at each alignment position is converted to a
Grantham Variation score (GV) and the difference between these properties and
those of the variant amino acid being assessed are calculated and a Grantham
Difference score generated (GD).  The predicted effect is classed as C0, C15,
C25, C35, C45, C55, or C65, with C65 most likely to interfere with function
and C0 least likely."><meta property="og:image" content="https://maialab.org/agvgd/logo.svg"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">agvgd</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/maialab/agvgd/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Align-GVGD (A-GVGD)</h1>
      <small class="dont-index">Source: <a href="https://github.com/maialab/agvgd/blob/HEAD/R/agvgd.R" class="external-link"><code>R/agvgd.R</code></a></small>
      <div class="d-none name"><code>agvgd.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function implements the Align-GVGD (A-GVGD) method described in
Tavtigian <em>et al.</em> (2006).</p>
<p>A-GVGD combines multiple sequence alignment of orthologous sequences with the
Grantham distance to classify missense variants, i.e. to distinguish human
disease susceptibility missense changes from changes of little clinical
significance.</p>
<p>The biochemical variation at each alignment position is converted to a
Grantham Variation score (GV) and the difference between these properties and
those of the variant amino acid being assessed are calculated and a Grantham
Difference score generated (GD).  The predicted effect is classed as C0, C15,
C25, C35, C45, C55, or C65, with C65 most likely to interfere with function
and C0 least likely.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">agvgd</span><span class="op">(</span><span class="va">alignment</span>, <span class="va">poi</span>, <span class="va">sub</span>, mode <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"recycle"</span>, <span class="st">"expand_grid"</span><span class="op">)</span>, sort <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>alignment</dt>
<dd><p>A character matrix or an alignment object obtained with
<code><a href="read_alignment.html">read_alignment()</a></code>. Rows are expected to be sequences of single characters
(protein residues), and columns the alignment positions. The first row must
be the reference sequence, i.e. the sequence whose substitutions will be
evaluated against.</p></dd>
<dt>poi</dt>
<dd><p>A whole number indicating the position of interest (POI).</p></dd>
<dt>sub</dt>
<dd><p>A character vector of protein residue substitutions to be
classified. The amino acids must be provided as one-letter symbols.</p></dd>
<dt>mode</dt>
<dd><p>If both <code>poi</code> and <code>sub</code> contain more than one element, <code>mode</code>
specifies how these two inputs are combined. If <code>mode = 'recycle'</code> the
shortest vector is recycled to match the length of the longest. If <code>mode = 'expand_grid'</code>, all combinations between elements of <code>poi</code> and <code>sub</code> are
combined.</p></dd>
<dt>sort</dt>
<dd><p>Whether to sort the output by <code>gd</code>, or not. Default is <code>FALSE</code>.</p></dd>
</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A <a href="https://tibble.tidyverse.org/reference/tibble-package.html" class="external-link">tibble</a> whose observations refer to the
combination alignment position and amino acid substitution; consists of
seven variables:</p><dl><dt>res</dt>
<dd><p>Position of the amino acid residue in the reference protein
(first sequence in the alignment). This position corresponds to <code>poi</code> minus
the gaps in the alignment.</p></dd>

<dt>poi</dt>
<dd><p>Position of interest, i.e. the alignment position at which the
amino acid substitution is being assessed.</p></dd>

<dt>ref</dt>
<dd><p>Reference amino acid, i.e. the amino acid in the first sequence
of the alignment, at the position of interest.</p></dd>

<dt>sub</dt>
<dd><p>Amino acid substitution being assessed.</p></dd>

<dt>gv</dt>
<dd><p>Grantham variation score.</p></dd>

<dt>gd</dt>
<dd><p>Grantham difference score.</p></dd>

<dt>prediction</dt>
<dd><p>Predicted effect of the amino acid substitution. This is
classed as C0, C15, C25, C35, C45, C55, or C65, with C65 most likely to
interfere with function and C0 least likely.</p></dd>


</dl></div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    
<ul><li><p>Tavtigian, S.V., Deffenbaugh, A. M., Yin, L., Judkins, T., Scholl, T.,
Samollow, P.B., de Silva, D., Zharkikh, A., Thomas, A. <em>Comprehensive
statistical study of 452 BRCA1 missense substitutions with classification of
eight recurrent substitutions as neutral</em>. Journal of Medical Genetics 43,
295--305 (2006). doi: <a href="https://doi.org/10.1136/jmg.2005.033878" class="external-link">10.1136/jmg.2005.033878</a>
.</p></li>
<li><p>Mathe, E., Olivier, M., Kato, S., Ishioka, C., Hainaut, P., Tavtigian, S.V.
<em>Computational approaches for predicting the biological effect of p53
missense mutations: a comparison of three sequence analysis based methods</em>.
Nucleic Acids Research 34, 1317--1325 (2006). doi: <a href="https://doi.org/10.1093/nar/gkj518" class="external-link">10.1093/nar/gkj518</a>
.</p></li>
</ul></div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="co"># Read an alignment into R, e.g. the alignment for gene ATM.</span></span>
<span class="r-in"><span class="va">alignment_ATM</span> <span class="op">&lt;-</span> <span class="fu"><a href="read_alignment.html">read_alignment</a></span><span class="op">(</span>gene <span class="op">=</span> <span class="st">'ATM'</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># Predict the impact of changing the first residue (Met) to a Serine (S).</span></span>
<span class="r-in"><span class="fu">agvgd</span><span class="op">(</span>alignment <span class="op">=</span> <span class="va">alignment_ATM</span>, poi <span class="op">=</span> <span class="fl">1</span>, sub <span class="op">=</span> <span class="st">'S'</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 1 × 7</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     res   poi ref   sub      gv    gd prediction</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span>     1     1 M     S         0  135. C65       </span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># `poi` can be a vector of positions, e.g., 3 thru 10, allow for prediction</span></span>
<span class="r-in"><span class="co"># of multiple positions at once.</span></span>
<span class="r-in"><span class="fu">agvgd</span><span class="op">(</span>alignment <span class="op">=</span> <span class="va">alignment_ATM</span>, poi <span class="op">=</span> <span class="fl">3</span><span class="op">:</span><span class="fl">10</span>, sub <span class="op">=</span> <span class="st">'S'</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 8 × 7</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     res   poi ref   sub      gv     gd prediction</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span>     2     3 S     S      87.4   0    C0        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">2</span>     3     4 L     S      31.8 123.   C35       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">3</span>     4     5 V     S      96.2  99.1  C15       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">4</span>     5     6 L     S      92.4  57.7  C0        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">5</span>     6     7 N     S     108.    6.20 C0        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">6</span>     7     8 D     S     223.   50.2  C0        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">7</span>     8     9 L     S       0   144.   C65       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">8</span>     9    10 L     S     102.   95.9  C15       </span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># `poi` expects a position in the frame of reference of the alignment, i.e.</span></span>
<span class="r-in"><span class="co"># an alignment position (a column index). However, if you know instead</span></span>
<span class="r-in"><span class="co"># the residue position in the reference sequence (first sequence in the</span></span>
<span class="r-in"><span class="co"># alignment), then you may use the function `residue_pos_to_alignment_pos()`</span></span>
<span class="r-in"><span class="co"># to convert from residue position to alignment position.</span></span>
<span class="r-in"><span class="co">#</span></span>
<span class="r-in"><span class="co"># Example: The second residue in the reference sequence of the ATM alignment</span></span>
<span class="r-in"><span class="co"># is a Serine, after a Methionine. In the alignment, there is a gap between</span></span>
<span class="r-in"><span class="co"># the two residues, so the alignment is 3 but the residue position on the</span></span>
<span class="r-in"><span class="co"># protein is 2.</span></span>
<span class="r-in"><span class="op">(</span><span class="va">poi2</span> <span class="op">&lt;-</span> <span class="fu"><a href="res_to_poi.html">res_to_poi</a></span><span class="op">(</span><span class="va">alignment_ATM</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 3</span>
<span class="r-in"><span class="fu">agvgd</span><span class="op">(</span>alignment <span class="op">=</span> <span class="va">alignment_ATM</span>, poi <span class="op">=</span> <span class="va">poi2</span>, sub <span class="op">=</span> <span class="st">'A'</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 1 × 7</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     res   poi ref   sub      gv    gd prediction</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span>     2     3 S     A      87.4  65.6 C0        </span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># Because changes are context-dependent, i.e. they depend on the residue</span></span>
<span class="r-in"><span class="co"># variation observed at a given alignment position, the same reference</span></span>
<span class="r-in"><span class="co"># residue when replaced with the same substitution will in general have</span></span>
<span class="r-in"><span class="co"># a different predicted impact.</span></span>
<span class="r-in"><span class="fu">agvgd</span><span class="op">(</span>alignment <span class="op">=</span> <span class="va">alignment_ATM</span>, poi <span class="op">=</span> <span class="fl">9</span><span class="op">:</span><span class="fl">10</span>, sub <span class="op">=</span> <span class="st">'S'</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 2 × 7</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     res   poi ref   sub      gv    gd prediction</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span>     8     9 L     S        0  144.  C65       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">2</span>     9    10 L     S      102.  95.9 C15       </span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># Use the ancillary function `amino_acids()` to get a vector of one-letter</span></span>
<span class="r-in"><span class="co"># residue substitutions if you want to quickly assess the impact of all</span></span>
<span class="r-in"><span class="co"># possible substitutions.</span></span>
<span class="r-in"><span class="fu">agvgd</span><span class="op">(</span>alignment <span class="op">=</span> <span class="va">alignment_ATM</span>, poi <span class="op">=</span> <span class="fl">1</span>, sub <span class="op">=</span> <span class="fu"><a href="amino_acids.html">amino_acids</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 20 × 7</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      res   poi ref   sub      gv    gd prediction</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span>     1     1 M     S         0 135.  C65       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span>     1     1 M     R         0  91.6 C65       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span>     1     1 M     L         0  14.3 C0        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span>     1     1 M     P         0  86.6 C65       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span>     1     1 M     T         0  81.0 C65       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span>     1     1 M     A         0  84.4 C65       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span>     1     1 M     V         0  21.5 C15       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span>     1     1 M     G         0 127.  C65       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span>     1     1 M     I         0  10.1 C0        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span>     1     1 M     F         0  28.5 C25       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">11</span>     1     1 M     Y         0  35.2 C35       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">12</span>     1     1 M     C         0 196.  C65       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">13</span>     1     1 M     H         0  86.3 C65       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">14</span>     1     1 M     Q         0 101.  C65       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">15</span>     1     1 M     N         0 141.  C65       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">16</span>     1     1 M     K         0  94.5 C65       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">17</span>     1     1 M     D         0 160.  C65       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">18</span>     1     1 M     E         0 126.  C65       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">19</span>     1     1 M     M         0   0   C0        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">20</span>     1     1 M     W         0  66.6 C65       </span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># Parameter `mode` gives you flexibility on how to combine `poi` and `sub`.</span></span>
<span class="r-in"><span class="fu">agvgd</span><span class="op">(</span>alignment <span class="op">=</span> <span class="va">alignment_ATM</span>, poi <span class="op">=</span> <span class="fl">3</span><span class="op">:</span><span class="fl">4</span>, sub <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'A'</span>, <span class="st">'V'</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 2 × 7</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     res   poi ref   sub      gv    gd prediction</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span>     2     3 S     A      87.4  65.6 C0        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">2</span>     3     4 L     V      31.8   0   C0        </span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># Use 'expand_grid' for all combinations.</span></span>
<span class="r-in"><span class="fu">agvgd</span><span class="op">(</span>alignment <span class="op">=</span> <span class="va">alignment_ATM</span>, poi <span class="op">=</span> <span class="fl">3</span><span class="op">:</span><span class="fl">4</span>, sub <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'A'</span>, <span class="st">'V'</span><span class="op">)</span>, mode <span class="op">=</span> <span class="st">'expand_grid'</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 4 × 7</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     res   poi ref   sub      gv    gd prediction</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span>     2     3 S     A      87.4  65.6 C0        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">2</span>     2     3 S     V      87.4  82.7 C15       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">3</span>     3     4 L     A      31.8  64.4 C25       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">4</span>     3     4 L     V      31.8   0   C0        </span>
<span class="r-in"></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by <a href="https://rmagno.eu/" class="external-link">Ramiro Magno</a>, <a href="http://iduarte.eu/" class="external-link">Isabel Duarte</a>, Ana-Teresa Maia, <a href="https://cintesis.eu/" class="external-link"><img src="https://maialab.org/agvgd/reference/figures/cintesis.svg" height="24"></a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

    </footer></div>

  

  

  </body></html>

