<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="agvgd">
<title>PIK3CA mutations impact • agvgd</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="PIK3CA mutations impact">
<meta property="og:description" content="agvgd">
<meta property="og:image" content="https://maialab.org/agvgd/logo.svg">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">agvgd</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/correia_magno2022.html">PIK3CA mutations impact</a>
    <a class="dropdown-item" href="../articles/lee2010.html">Missense Variations in the BRCT Domain of BRCA1</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/maialab/agvgd/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>PIK3CA mutations impact</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/maialab/agvgd/blob/HEAD/vignettes/correia_magno2022.Rmd" class="external-link"><code>vignettes/correia_magno2022.Rmd</code></a></small>
      <div class="d-none name"><code>correia_magno2022.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>PIK3CA is a frequently mutated oncogene in human cancers, with
mutations found in various cancer types, including endometrial, ovarian,
colorectal, breast, cervical, squamous cell cancer of the head and neck,
chondroma, and thyroid carcinoma<span class="citation"><sup>1</sup></span>. The PIK3CA gene encodes the p110α
catalytic subunit of phosphatidylinositol 3-kinase<span class="citation"><sup>2</sup></span>.</p>
<p>Somatic missense mutations in PIK3CA have been reported in many human
cancer types, and are predominantly found in the kinase and helical
domains of the PIK3CA subunit<span class="citation"><sup>3</sup></span>.</p>
<p>According to karakas_2006 at al, some of the most common PIK3CA
mutations associated with cancer include:</p>
<ol style="list-style-type: decimal">
<li>Exon 9 (G1624A:E542K)</li>
<li>Exon 20 (A3140G:H1047R)</li>
<li>Exon 1 (E545K)</li>
</ol>
<p>PIK3CA mutations are more common in tumors with mutant PTEN genes
than in tumors with wild-type PTEN genes<span class="citation"><sup>4</sup></span>. The discovery of oncogenic
mutations in PIK3CA has emphasized the important role of PI3K in cancer
pathogenesis and made it possible to quickly and easily identify tumors
with activation of PI3K signaling by virtue of mutations in PIK3CA<span class="citation"><sup>4</sup></span>.</p>
<p>In this vignette we use <a href="https://maialab.org/agvgd/">agvgd</a> to classify the
pathogenecity of the missence mutations/variations reported in Correia
and Magno et al (2022)<span class="citation"><sup>5</sup></span>. Here
they report several mutations found in breast cancer, from which we
selected the 17 variants found in the metabric dataset. These will be
used to showcase how the usage of the <a href="https://maialab.org/agvgd/">agvgd</a> package can
complement the findings from other biomedical data analyses.</p>
<p><img src="../reference/figures/correia_magno2022_fig02.png" width="90%"></p>
</div>
<div class="section level2">
<h2 id="running-agvgd">Running agvgd<a class="anchor" aria-label="anchor" href="#running-agvgd"></a>
</h2>
<div class="section level3">
<h3 id="data-sets">Data sets<a class="anchor" aria-label="anchor" href="#data-sets"></a>
</h3>
<p>To use the <a href="https://maialab.org/agvgd/">agvgd</a> package to predict the protein impact
of the metabric variants reported in Correia and Magno et al<span class="citation"><sup>5</sup></span>, we need two data sets:</p>
<ul>
<li>The list of the 17 metabric PIK3CA missense variants;</li>
<li>The protein sequence alignment of PIK3CA.</li>
</ul>
<p>These first dataset is readily available with the
<a href="https://maialab.org/agvgd/">agvgd</a> package, and the protein alignment is provide
within the companion package <a href="https://github.com/maialab/protean" class="external-link">protean</a>. More information
regarding the <a href="https://github.com/maialab/protean" class="external-link">protean</a> package can be found in its <a href="https://github.com/maialab/protean" class="external-link">GitHub repository</a>.</p>
<p><br></p>
<div class="section level4">
<h4 id="get-the-alignment-dataset-from-protean">1. Get the alignment dataset from <code>{protean}</code><a class="anchor" aria-label="anchor" href="#get-the-alignment-dataset-from-protean"></a>
</h4>
<p>To retrieve the alignment of PIK3CA from the <a href="https://github.com/maialab/protean" class="external-link">protean</a>
package, we will use the functions <code><a href="https://rdrr.io/pkg/protean/man/read_profile.html" class="external-link">read_profile()</a></code> and
<code><a href="https://rdrr.io/pkg/protean/man/profile_path.html" class="external-link">profile_path()</a></code> using the gene name
<code>"PIK3CA"</code>:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load the required packages</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/maialab/protean" class="external-link">protean</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Using protean</span></span>
<span></span>
<span>  <span class="co"># Check all oncoKB proteins that have alignment files in the protean package</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu">protean</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/protean/man/oncokb_genes.html" class="external-link">oncokb_genes</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "ABL1"  "AKT1"  "ALK"   "AMER1" "APC"   "AR"</span></span>
<span></span>
<span>  <span class="co"># Confirm that PIK3CA is present</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"PIK3CA"</span>, <span class="fu">protean</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/protean/man/oncokb_genes.html" class="external-link">oncokb_genes</a></span>, value <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "PIK3CA"</span></span>
<span></span>
<span>  <span class="co"># Get the alignment profile information for PIK3CA</span></span>
<span>  <span class="va">pik3ca_profile</span> <span class="op">&lt;-</span> <span class="fu">protean</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/protean/man/read_profile.html" class="external-link">read_profile</a></span><span class="op">(</span><span class="fu">protean</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/protean/man/profile_path.html" class="external-link">profile_path</a></span><span class="op">(</span><span class="st">"PIK3CA"</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">pik3ca_profile</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 11</span></span></span>
<span><span class="co">#&gt;   timestamp            human_prot_id ortho_prot_id ortho_species human_align_seq</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> 2023-11-26 21:55:44… ENSP00000263… ENSGGOP00000… gorilla_gori… MPPRPSSGELWGIH…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> 2023-11-26 21:55:44… ENSP00000263… ENSMFAP00000… macaca_fasci… MPPRPSSGELWGIH…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> 2023-11-26 21:55:44… ENSP00000263… ENSMMUP00000… macaca_mulat… MPPRPSSGELWGIH…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> 2023-11-26 21:55:44… ENSP00000263… ENSMNEP00000… macaca_nemes… MPPRPSSGELWGIH…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> 2023-11-26 21:55:44… ENSP00000263… ENSPANP00000… papio_anubis  MPPRPSSGELWGIH…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> 2023-11-26 21:55:44… ENSP00000263… ENSMLEP00000… mandrillus_l… MPPRPSSGELWGIH…</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 6 more variables: ortho_align_seq &lt;chr&gt;, human_ortho_perc_id &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ortho_human_perc_id &lt;dbl&gt;, cigar &lt;chr&gt;, human_profile_seq &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ortho_profile_seq &lt;chr&gt;</span></span></span></code></pre></div>
<p><br></p>
</div>
<div class="section level4">
<h4 id="format-the-sequence-profile-to-an-alignment-with-agvgd">2. Format the sequence profile to an alignment with
<code>{agvgd}</code><a class="anchor" aria-label="anchor" href="#format-the-sequence-profile-to-an-alignment-with-agvgd"></a>
</h4>
<p>Next, the sequence profile must be converted to an alignment format.
For this we must use the <code><a href="../reference/profile_to_alignment.html">profile_to_alignment()</a></code> function
from the <a href="https://maialab.org/agvgd/">agvgd</a> package.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># Using agvgd </span></span>
<span></span>
<span><span class="co"># Convert the protean sequence profile to an alignment</span></span>
<span><span class="va">pik3ca_alignment</span> <span class="op">&lt;-</span> <span class="fu">agvgd</span><span class="fu">::</span><span class="fu"><a href="../reference/profile_to_alignment.html">profile_to_alignment</a></span><span class="op">(</span><span class="va">pik3ca_profile</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Check the first lines and columns of the alignment</span></span>
<span><span class="va">pik3ca_alignment</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">15</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">30</span><span class="op">]</span></span>
<span><span class="co">#&gt; homo_sapiens_ENSP00000263967               1 MPPRPSSGELWGIHLMPPRILVECLLPNGM</span></span>
<span><span class="co">#&gt; gorilla_gorilla_ENSGGOP00000009158         1 MPPRPSSGELWGIHLMPPRILVECLLPNGM</span></span>
<span><span class="co">#&gt; macaca_fascicularis_ENSMFAP00000027509     1 MPPRPSSGELWGIHLMPPRILVECLLPNGM</span></span>
<span><span class="co">#&gt; macaca_mulatta_ENSMMUP00000006866          1 MPPRPSSGELWGIHLMPPRILVECLLPNGM</span></span>
<span><span class="co">#&gt; macaca_nemestrina_ENSMNEP00000020318       1 MPPRPSSGELWGIHLMPPRILVECLLPNGM</span></span>
<span><span class="co">#&gt; papio_anubis_ENSPANP00000023010            1 MPPRPSSGELWGIHLMPPRILVECLLPNGM</span></span>
<span><span class="co">#&gt; mandrillus_leucophaeus_ENSMLEP00000019594  1 MPPRPSSGELWGIHLMPPRILVECLLPNGM</span></span>
<span><span class="co">#&gt; chlorocebus_sabaeus_ENSCSAP00000009123     1 MPPRPSSGELWGIHLMPPRILVECLLPNGM</span></span>
<span><span class="co">#&gt; pongo_abelii_ENSPPYP00000041486            1 MPPRPSSGELWGIHLMPPRILVECLLPNGM</span></span>
<span><span class="co">#&gt; rhinopithecus_roxellana_ENSRROP00000011898 1 MPPRPSSGELWGIHLMPPRILVECLLPNGM</span></span>
<span><span class="co">#&gt; equus_asinus_ENSEASP00005039543            1 MPPRPSSGELWGIHLMPPRILVECLLPNGM</span></span>
<span><span class="co">#&gt; capra_hircus_ENSCHIP00000015546            1 MPPRPSSGELWGIHLMPPRILVECLLPNGM</span></span>
<span><span class="co">#&gt; panthera_leo_ENSPLOP00000019775            1 MPPRPSSGELWGIHLMPPRILVECLLPNGM</span></span>
<span><span class="co">#&gt; ailuropoda_melanoleuca_ENSAMEP00000011350  1 MPPRPSSGELWGIHLMPPRILVECLLPNGM</span></span>
<span><span class="co">#&gt; bos_indicus_hybrid_ENSBIXP00005013124      1 MPPRPSSGELWGIHLMPPRILVECLLPNGM</span></span></code></pre></div>
<p><br></p>
</div>
<div class="section level4">
<h4 id="read-the-file-with-substitutions-of-interest">3. Read the file with substitutions of interest<a class="anchor" aria-label="anchor" href="#read-the-file-with-substitutions-of-interest"></a>
</h4>
<p>The substitutions file should have one substitution per line, and the
substitutions should be formatted in the following way:</p>
<p><strong>H1047Y</strong> where H (Histidine) is the original
aminoacid, 1047 is the protein position for this aminoacid, and Y
(Tyrosine) is the new aminoacid originated by the missence mutation.</p>
<p>To read in the 17 PIK3CA missense variants, we use the function
<code><a href="../reference/read_substitutions.html">read_substitutions()</a></code> from the <a href="https://maialab.org/agvgd/">agvgd</a>
package:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Read the variants file</span></span>
<span><span class="va">path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/correia_magno2022.txt"</span>, package <span class="op">=</span> <span class="st">'agvgd'</span>, mustWork <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">missense_variants</span> <span class="op">&lt;-</span> <span class="fu">agvgd</span><span class="fu">::</span><span class="fu"><a href="../reference/read_substitutions.html">read_substitutions</a></span><span class="op">(</span><span class="va">path</span><span class="op">)</span></span>
<span><span class="va">missense_variants</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 17 × 4</span></span></span>
<span><span class="co">#&gt;      res   poi ref   sub  </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span>    88    <span style="color: #BB0000;">NA</span> R     Q    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span>   111    <span style="color: #BB0000;">NA</span> K     E    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span>   345    <span style="color: #BB0000;">NA</span> N     K    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span>   385    <span style="color: #BB0000;">NA</span> E     K    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span>   420    <span style="color: #BB0000;">NA</span> C     R    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span>   453    <span style="color: #BB0000;">NA</span> E     K    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span>   542    <span style="color: #BB0000;">NA</span> E     K    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span>   545    <span style="color: #BB0000;">NA</span> E     A    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span>   545    <span style="color: #BB0000;">NA</span> E     K    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span>   546    <span style="color: #BB0000;">NA</span> Q     R    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span>   726    <span style="color: #BB0000;">NA</span> E     K    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">12</span>   909    <span style="color: #BB0000;">NA</span> F     L    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">13</span>  <span style="text-decoration: underline;">1</span>021    <span style="color: #BB0000;">NA</span> Y     H    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">14</span>  <span style="text-decoration: underline;">1</span>043    <span style="color: #BB0000;">NA</span> M     V    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">15</span>  <span style="text-decoration: underline;">1</span>047    <span style="color: #BB0000;">NA</span> H     L    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">16</span>  <span style="text-decoration: underline;">1</span>047    <span style="color: #BB0000;">NA</span> H     R    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">17</span>  <span style="text-decoration: underline;">1</span>047    <span style="color: #BB0000;">NA</span> H     Y</span></span></code></pre></div>
<p>The list of missense variants comes with the indication of the
residue number in the protein sequence, i.e. column <code>res</code> in
<code>missense_variants</code>. As we’ll see next, the function
<code><a href="../reference/agvgd.html">agvgd()</a></code> uses the alignment positions (column
<code>poi</code> in <code>missense_variants</code>) instead to refer to
those positions in the alignment.</p>
<p>The difference between <code>res</code> and <code>poi</code>
(position of interest) is that <code>res</code> counts the residues in
the protein primary sequence reference, while <code>poi</code> refers to
the positions in the alignment, accounting for gaps.</p>
<p>So, before we proceed, we will update the
<code>missense_variants</code> tibble and replace the <code>NA</code>
values with the corresponding <code>poi</code> values. For that we use
the function <code><a href="../reference/res_to_poi.html">res_to_poi()</a></code> from <a href="https://maialab.org/agvgd/">agvgd</a>:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">missense_variants</span><span class="op">$</span><span class="va">poi</span> <span class="op">&lt;-</span> <span class="fu">agvgd</span><span class="fu">::</span><span class="fu"><a href="../reference/res_to_poi.html">res_to_poi</a></span><span class="op">(</span><span class="va">pik3ca_alignment</span>, <span class="va">missense_variants</span><span class="op">$</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">missense_variants</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 4</span></span></span>
<span><span class="co">#&gt;     res   poi ref   sub  </span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span>    88    88 R     Q    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span>   111   111 K     E    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span>   345   345 N     K    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span>   385   385 E     K    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span>   420   420 C     R    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span>   453   453 E     K</span></span></code></pre></div>
<p><br></p>
</div>
<div class="section level4">
<h4 id="calculate-the-scores-with-agvgd">4. Calculate the scores with <em>agvgd</em><a class="anchor" aria-label="anchor" href="#calculate-the-scores-with-agvgd"></a>
</h4>
<p>Run Align-GVGD algorithm with the function <code><a href="../reference/agvgd.html">agvgd()</a></code>:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">scores</span> <span class="op">&lt;-</span> <span class="fu">agvgd</span><span class="fu">::</span><span class="fu"><a href="../reference/agvgd.html">agvgd</a></span><span class="op">(</span>alignment <span class="op">=</span> <span class="va">pik3ca_alignment</span>,</span>
<span>      poi <span class="op">=</span> <span class="va">missense_variants</span><span class="op">$</span><span class="va">poi</span>,</span>
<span>      sub <span class="op">=</span> <span class="va">missense_variants</span><span class="op">$</span><span class="va">sub</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">scores</span>, n <span class="op">=</span> <span class="cn">Inf</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 17 × 7</span></span></span>
<span><span class="co">#&gt;      res   poi ref   sub      gv    gd prediction</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span>    88    88 R     Q      354.     0 C0        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span>   111   111 K     E      354.     0 C0        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span>   345   345 N     K      354.     0 C0        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span>   385   385 E     K      354.     0 C0        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span>   420   420 C     R      354.     0 C0        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span>   453   453 E     K      354.     0 C0        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span>   542   542 E     K      354.     0 C0        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span>   545   545 E     A      354.     0 C0        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span>   545   545 E     K      354.     0 C0        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span>   546   546 Q     R      354.     0 C0        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span>   726   726 E     K      354.     0 C0        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">12</span>   909   909 F     L      354.     0 C0        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">13</span>  <span style="text-decoration: underline;">1</span>021  <span style="text-decoration: underline;">1</span>021 Y     H      354.     0 C0        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">14</span>  <span style="text-decoration: underline;">1</span>043  <span style="text-decoration: underline;">1</span>043 M     V      354.     0 C0        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">15</span>  <span style="text-decoration: underline;">1</span>047  <span style="text-decoration: underline;">1</span>047 H     L      354.     0 C0        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">16</span>  <span style="text-decoration: underline;">1</span>047  <span style="text-decoration: underline;">1</span>047 H     R      354.     0 C0        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">17</span>  <span style="text-decoration: underline;">1</span>047  <span style="text-decoration: underline;">1</span>047 H     Y      354.     0 C0</span></span></code></pre></div>
<p><br></p>
<div class="section level5">
<h5 id="critical-analysis">Critical analysis<a class="anchor" aria-label="anchor" href="#critical-analysis"></a>
</h5>
<p>These results show that using all the sequences present in the
ENSEMBL orthologous database will not provide insightful information for
agvgd, since this method is very sensitive to the alignment. Here, it
has classified all the variants with a score of C0, meaning that none of
the changes of interest are clinically relevant.<br>
This is because using sequences that are so distantly related in the
tree of life it is no longer reasonable to assume that those distant
orthologous proteins share the same functions as the human protein of
interest. Accordingly, the alignment must be curated to include a
reasonable amount of sequence variation, but keeping only a subset of
the proteins that are still evolutionarily close enough, so that they
still share the same function, and therefore the same sequence
constraints.</p>
<p><br></p>
</div>
</div>
<div class="section level4">
<h4 id="curate-the-alignment-and-rerun-the-score-calculation">5. Curate the alignment and rerun the score calculation<a class="anchor" aria-label="anchor" href="#curate-the-alignment-and-rerun-the-score-calculation"></a>
</h4>
<p>A first round of curation can be done automatically by removing
sequences so distant that they are over 90% different from the protein
of interest (bidirectionally, i.e., between human and the orthologous,
and between the orthologous and the human).</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># Remove distant orthologs and redundant seqs</span></span>
<span><span class="va">pik3ca_profile</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">human_ortho_perc_id</span> <span class="op">&gt;=</span> <span class="fl">90</span> <span class="op">&amp;</span> <span class="va">ortho_human_perc_id</span> <span class="op">&gt;=</span> <span class="fl">90</span><span class="op">)</span> <span class="op">-&gt;</span> <span class="va">profile2</span></span>
<span></span>
<span><span class="co"># Format the profile to an alignment</span></span>
<span><span class="va">alignment2</span> <span class="op">&lt;-</span> <span class="fu">agvgd</span><span class="fu">::</span><span class="fu"><a href="../reference/profile_to_alignment.html">profile_to_alignment</a></span><span class="op">(</span><span class="va">profile2</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Calculate the scores for the filtered alignment</span></span>
<span><span class="va">scores2</span> <span class="op">&lt;-</span> <span class="fu">agvgd</span><span class="fu">::</span><span class="fu"><a href="../reference/agvgd.html">agvgd</a></span><span class="op">(</span>alignment <span class="op">=</span> <span class="va">alignment2</span>,</span>
<span>      poi <span class="op">=</span> <span class="va">missense_variants</span><span class="op">$</span><span class="va">poi</span>,</span>
<span>      sub <span class="op">=</span> <span class="va">missense_variants</span><span class="op">$</span><span class="va">sub</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">scores2</span>, n <span class="op">=</span> <span class="cn">Inf</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 17 × 7</span></span></span>
<span><span class="co">#&gt;      res   poi ref   sub      gv     gd prediction</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span>    88    88 R     Q       0    42.8  C35       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span>   111   111 K     E       0    56.9  C55       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span>   345   345 N     K       0    93.9  C65       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span>   385   385 E     K     223.   39.9  C0        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span>   420   420 C     R       0   180.   C65       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span>   453   453 E     K     223.   39.9  C0        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span>   542   542 E     K      56.9   0    C0        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span>   545   545 E     A       0   107.   C65       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span>   545   545 E     K       0    56.9  C55       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span>   546   546 Q     R       0    42.8  C35       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span>   726   726 E     K     239.    0    C0        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">12</span>   909   909 F     L       0    21.8  C15       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">13</span>  <span style="text-decoration: underline;">1</span>021  <span style="text-decoration: underline;">1</span>021 Y     H     354.    0    C0        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">14</span>  <span style="text-decoration: underline;">1</span>043  <span style="text-decoration: underline;">1</span>043 M     V     232.    3.24 C0        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">15</span>  <span style="text-decoration: underline;">1</span>047  <span style="text-decoration: underline;">1</span>047 H     L     260.    4.86 C0        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">16</span>  <span style="text-decoration: underline;">1</span>047  <span style="text-decoration: underline;">1</span>047 H     R     260.    1.62 C0        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">17</span>  <span style="text-decoration: underline;">1</span>047  <span style="text-decoration: underline;">1</span>047 H     Y     260.    4.05 C0</span></span></code></pre></div>
<p><br></p>
<div class="section level5">
<h5 id="critical-analysis-after-alignment-filtering">Critical analysis after alignment filtering<a class="anchor" aria-label="anchor" href="#critical-analysis-after-alignment-filtering"></a>
</h5>
<p>After removal of distantly related proteins, the results show enough
variability to allow the classification of variants that are know to be
clinically relevant (for example, E545K with C65 score), showing the
relevance of this step for the accuracy of the algorithm.</p>
<p>Additional rounds of curation can be done manually, by visually
inspecting the alignment (for example using <a href="https://www.jalview.org/" class="external-link">Jalview</a>), and re-running the
<code><a href="../reference/agvgd.html">agvgd()</a></code> score calculation function using the manually
curated alignments as input.</p>
<p><br></p>
</div>
</div>
</div>
<div class="section level3 unnumbered">
<h3 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h3>
<div id="refs" class="references csl-bib-body" line-spacing="2">
<div id="ref-german_2013" class="csl-entry">
<div class="csl-left-margin">1. </div>
<div class="csl-right-inline">German, S. <em>et al.</em> <a href="https://doi.org/10.1186/1897-4287-11-5" class="external-link">Carcinogenesis of
<span>PIK3CA</span></a>. <em>Hereditary Cancer in Clinical Practice</em>
<strong>11</strong>, 5 (2013).</div>
</div>
<div id="ref-ligresti_2009" class="csl-entry">
<div class="csl-left-margin">2. </div>
<div class="csl-right-inline">Ligresti, G. <em>et al.</em> <a href="https://doi.org/10.4161/cc.8.9.8255" class="external-link"><span>PIK3CA</span> mutations
in human solid tumors: <span>Role</span> in sensitivity to various
therapeutic approaches</a>. <em>Cell Cycle</em> <strong>8</strong>,
1352–1358 (2009).</div>
</div>
<div id="ref-karakas_2006" class="csl-entry">
<div class="csl-left-margin">3. </div>
<div class="csl-right-inline">Karakas, B., Bachman, K. E. &amp; Park, B. H.
<a href="https://doi.org/10.1038/sj.bjc.6602970" class="external-link">Mutation of the PIK3CA
oncogene in human cancers</a>. <em>British Journal of Cancer</em>
<strong>94</strong>, 455–459 (2006).</div>
</div>
<div id="ref-samuels_2010" class="csl-entry">
<div class="csl-left-margin">4. </div>
<div class="csl-right-inline">Samuels, Y. &amp; Waldman, T. <a href="https://doi.org/10.1007/82_2010_68" class="external-link">Oncogenic
<span>Mutations</span> of <span>PIK3CA</span> in <span>Human</span>
<span>Cancers</span></a>. in <em>Phosphoinositide 3-kinase in
<span>Health</span> and <span>Disease</span></em> (eds. Rommel, C.,
Vanhaesebroeck, B. &amp; Vogt, P. K.) vol. 347 21–41 (Springer Berlin
Heidelberg, 2010).</div>
</div>
<div id="ref-correia_magno_2022" class="csl-entry">
<div class="csl-left-margin">5. </div>
<div class="csl-right-inline">Correia, L. <em>et al.</em> <a href="https://doi.org/10.1038/s41523-022-00435-9" class="external-link">Allelic expression
imbalance of <span>PIK3CA</span> mutations is frequent in breast cancer
and prognostically significant</a>. <em>npj Breast Cancer</em>
<strong>8</strong>, 71 (2022).</div>
</div>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by <a href="https://rmagno.eu/" class="external-link">Ramiro Magno</a>, <a href="http://iduarte.eu/" class="external-link">Isabel Duarte</a>, Ana-Teresa Maia, <a href="https://cintesis.eu/" class="external-link"><img src="https://maialab.org/agvgd/reference/figures/cintesis.svg" height="24"></a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
